WIP: Feature - 2925 - Logged out page

esizer opened the pull request:
Resolves #2925 

## WIP

This blocked by a few other issues so just waiting on those to finish it off.

- #2924 Waiting on the implementation of the alert here
- #2923 This has some routing needed to display the page
- #2896 Auth container needed to display log out confirmation dialog

## Summary

This created a page at `/logged-out` to display to a user **_after_** they have logged out. If they land on this page while logged in, a dialog will appear asking if they would like to logout.

## New Component - `<TileLink />`

This is a simplified version of the `<Link />`  component that accepts the following:

- **color**: Color - _required_
  - This controls the colour of the border to the left
- **href**: string - _required_
  - URL to point the link to
-  **external**: boolean
  - If the link goes to external location, avoid using `universal-router`

petertgiles reviewed (changes requested):
So strictly speaking this is out of scope for this PR but it did happen in your PR from less than 24 hours ago so...
Could you please bump up the font weights in the header?  In Firefox-Linux (and I'm guessing Safari?) it becomes almost invisible.
![image](https://user-images.githubusercontent.com/8978655/173080620-4afdf5ce-8ae2-42cd-827c-00a90cb1ed08.png)

petertgiles commented:
Much better, thanks!
![image](https://user-images.githubusercontent.com/8978655/173082600-2840f68c-f04f-4b6a-a5cd-45bd4ce2b448.png)

esizer commented:
> Much better, thanks!

Still a bit lighter than the other two links and they are using the same component 😖

petertgiles reviewed (commented):
Some thoughts:
- You would need to add OAUTH_LOGOUT_URI to the talentsearch SJS page to make the post-logout redirect work properly, right?
- If I manually navigate to the /logged-out page to pop the dialog and click cancel I'm just ending up on /talentsearch.  Can we actually go back to the previous page?
- I think the difference in styling between Request and Login is that one is a link and one is a button.
- When you click logout there's a super simple browser dialog but on the logged-out page there's a super fancy react dialog.  Is there any advantage to picking one or the other for consistency?

esizer commented:
> > * I think the difference in styling between Request and Login is that one is a link and one is a button.
> 
> Woah, why are our links using open sans vs buttons using fira? I'm going to be honest, most fonts look the same to me and i never noticed this. I'll look into this. Is this intentional or should open sans be used everywhere?

Mentioned in slack, but this is due to an incomplete css reset. I have added a workaround for now in [add workaround font reset](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2992/commits/2a7e25f3d69a082a38a60a07ce643a72ac3c0431)

petertgiles commented:
> I'm going to be honest, that sounds correct but I did not and it worked for me during testing shrug . Did you need to do that to get it to work?

Yeah, I did.  I'm pretty confident that this is necessary

> So, I may be misunderstanding this. If you navigate directly to it that means typing it into the address bar, right? If so, the previous page only exists in browser history which we do not have access to.  The only thing I can think of is: window.history.goBack(). If they meant to go to talent.canda.ca from a different website, they would actually be directed back to that previous website and not to our application.

That's fair.  If a simple "back" is not possible then let's maybe send them back to the home page for the entire app "/"?

> Woah, why are our links using open sans vs buttons using fira? I'm going to be honest, most fonts look the same to me and i never noticed this. I'll look into this. Is this intentional or should open sans be used everywhere?

Sounds like you've got this one figured out.  I'm definitely not the expert on this one. :grin: 

> I like having consistency and prefer the react dialog. If you agree, I don't think it will be much to update them.

Sounds good to me!

petertgiles commented:
> I'm going to be honest, that sounds correct but I did not and it worked for me during testing shrug .

I have a theory about that... I actually had the opposite problem during testing.  I had the "disable cache" box selected but the update SJS file wouldn't get pulled in until I actually loaded that file directly in my browser!  I wonder if you had it cached with the logout URL populated even though Apache wouldn't have it.

esizer commented:
> Yeah, I did. I'm pretty confident that this is necessary

Fixed with [add logout uri to sjs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2992/commits/42dfbf8fd722fb729b7f58e208d59dd5790a703b)

> That's fair. If a simple "back" is not possible then let's maybe send them back to the home page for the entire app "/"?

Updated the cancel url in [change cancel link destination](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2992/commits/c63ea0ff0128027bac5e11439c4c17e230a20f8b)

> Sounds like you've got this one figured out. I'm definitely not the expert on this one. 😁

Add workaround in [add workaround font reset](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2992/commits/2a7e25f3d69a082a38a60a07ce643a72ac3c0431)

> Sounds good to me!

Updated admin and talentsearch logout dialogs in [update logout confirmation dialogs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2992/commits/5b99a19d1f1c03a42285f29dd18014ad18b8832e)
